### PR TITLE
OpenVPN bump to 2.3.12 and improved support for multi-instance setups

### PR DESCRIPTION
**Pull request by jimklimov**

### CODE REVIEW

File: components/network/openvpn/files/openvpn.xml [hunk 1]
```diff
@@ -1,8 +1,69 @@
 <?xml version="1.0" ?>
 <!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+
```
Comment (xen0l) [inline components/network/openvpn/files/openvpn.xml]: Move these bits to other location, e.g. /usr/share/doc/openvpn/README.openindiana and just reference that file here.

File: components/network/openvpn/files/openvpn.xml [hunk 2]
```diff
@@ -1,8 +1,69 @@
...
+You might also want to enable some permissions in the service manifest below,
+or per-instance, if e.g. you want one to listen on privileged ports, or to
+chroot(), or to use configuration files and device nodes owned by another user.
+Remember that under SMF a "root" is not almighty! For more details, see the
+commented "method_credential" suggestions below.
+-->
 <service_bundle type="manifest" name="openvpn">
-    <service name="network/openvpn" type="service" version="0">
-	<create_default_instance enabled="false"/>
+    <service name="network/openvpn" type="service" version="0.01">
```
Comment (xen0l) [inline components/network/openvpn/files/openvpn.xml]: Use 1, instead of 0.01

File: components/network/openvpn/files/openvpn.xml [hunk 3]
```diff
@@ -12,30 +14,51 @@
             <service_fmri value="svc:/system/filesystem/local"/>
         </dependency>
 
-        <method_context>
+        <method_context project=':default' resource_pool=':default' working_directory='/etc/openvpn'>
```
Comment (xen0l) [inline components/network/openvpn/files/openvpn.xml]: project and resource_pool are something we shouldn't user unless we have support for it in IPS. As for working directory, won't it cause the issue, if people use different location for config file?

File: components/network/openvpn/files/openvpn.xml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" ?>
 <!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<!-- See the non-trivial-setup notes and recommendations for the OpenVPN SMF
+     service integration in /usr/share/doc/openvpn/README.openindiana -->
 <service_bundle type="manifest" name="openvpn">
-    <service name="network/openvpn" type="service" version="0">
-	<create_default_instance enabled="false"/>
+    <service name="network/openvpn" type="service" version="0.01">
```
Comment (xen0l) [inline components/network/openvpn/files/openvpn.xml]: 0.01 -> 1

File: components/network/openvpn/openvpn.p5m [hunk 5]
```diff
@@ -25,6 +25,7 @@ license openvpn.license license='GPLv2'
 file files/auth_openvpn path=etc/security/auth_attr.d/openvpn
 file files/prof_openvpn path=etc/security/prof_attr.d/openvpn
 file files/openvpn.xml path=lib/svc/manifest/network/openvpn.xml restart_fmri=svc:/system/manifest-import:default
+file files/README.openindiana path=usr/share/doc/openvpn/README.openindiana
 
 dir  path=etc
```
Comment (xen0l) [inline components/network/openvpn/openvpn.p5m]: Can you add dir action for /var/log/openvpn and optionally remove other dir actions, so p5m file is consistent with others?

File: components/network/openvpn/files/openvpn.xml [hunk 6]
```diff
@@ -12,30 +14,51 @@
...
 
-	<property_group name='config' type='application'>
-	    <propval name="config_file" type='astring' value='/etc/openvpn/openvpn.conf'/>
-	    <propval name='value_authorization' type='astring' value='solaris.smf.value.openvpn'/>	
-	</property_group>
+        <property_group name='config' type='application'>
+            <!-- Full path to config file is required -->
+            <propval name="config_file" type='astring' value='/etc/openvpn/openvpn.conf'/>
+            <!-- Note that the log location can be overridden inside the config file -->
+            <propval name="default_logfile" type='astring' value='/dev/stderr'/>
```
Comment (xen0l) [inline components/network/openvpn/files/openvpn.xml]: What about using /var/log/openvpn/%i.log instead of /dev/stderr?

Comment (jimklimov) [reply components/network/openvpn/files/openvpn.xml]: not in the variable at least - so the string with macro will get hardcoded to exec line instead.. but doable...

Comment (jimklimov) [inline components/network/openvpn/files/openvpn.xml]: not in the variable at least - so the string with macro will get hardcoded to exec lien instead.. but doable...

File: components/network/openvpn/files/openvpn.xml (hunk 3 above)
Comment (jimklimov) [reply components/network/openvpn/files/openvpn.xml]: The workdir should not influence an explicitly specified "atomic" config file, and should help with files referenced from it by relative locations - if they are all in /etc/openvpn per defaults. Although with the other bit (--cd) - the command-line option wins.

**Comment by pyhalov** [comment]
@xenol, @jimklimov  - what's is the current status of this PR?

**Comment by jimklimov** [comment]
Never an end to perfection there is, but interruptions there may be :-)
Other ideas = other PRs? ;)

**Pull request merged**